modelkit/assets/manager.py

```diff
@@ -81,74 +81,79 @@ def __init__(
 
     def _fetch_asset(self, spec: AssetSpec, _force_download=False):
         with ContextualizedLogging(name=spec.name):
-            local_name = os.path.join(self.assets_dir, *spec.name.split("/"))
-            local_versions = spec.get_local_versions(local_name)
-            logger.debug("Local versions", local_versions=local_versions)
-
-            if spec.is_version_complete():
-                with ContextualizedLogging(version=spec.version):
-                    return self._fetch_asset_version(
-                        spec, local_versions, _force_download
-                    )
 
-            remote_versions = []
-            if self.storage_provider:
-                remote_versions = self.storage_provider.get_versions_info(spec.name)
-                logger.debug("Fetched remote versions", remote_versions=remote_versions)
+            self._resolve_version(spec)
```

Comment:
logic of resolution and download of assets were mixed and benefits from being split formally here.